internal/utils/labels.go

```diff
@@ -0,0 +1,31 @@
+package utils
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+func ConvertLabelsToMap(labelsString string) (map[string]string, error) {
+	result := map[string]string{}
+	splitFn := func(c rune) bool {
+		return c == ','
+	}
+	vals := strings.FieldsFunc(labelsString, splitFn)
+	for _, val := range vals {
+		pair := strings.SplitN(val, "=", 2)
+		if len(pair) < 2 {
+			return nil, fmt.Errorf("invalid value %s", val)
+		}
```

Comment:
Labels may have an empty value, so this should be handled as if the value was an empty string instead of an error.